src/compiler/scala/tools/nsc/transform/Fields.scala

```diff
@@ -83,7 +83,7 @@ abstract class Fields extends InfoTransform with ast.TreeDSL with TypingTransfor
   private def accessorImplementedInSubclass(accessor: Symbol) =
     (accessor hasFlag SYNTHESIZE_IMPL_IN_SUBCLASS) && (accessor hasFlag (ACCESSOR | MODULE))
 
-  private def concreteOrSynthImpl(sym: Symbol): Boolean = !(sym hasFlag DEFERRED) || (sym hasFlag SYNTHESIZE_IMPL_IN_SUBCLASS)
+  @inline final def notDeferredOrSynthImpl(sym: Symbol): Boolean = !(sym hasFlag DEFERRED) || (sym hasFlag SYNTHESIZE_IMPL_IN_SUBCLASS)
```

Comment:
because I added another layer [below in Mixin](https://github.com/scala/scala/pull/5206/files/c498000413303c837fa8c66865a47471b4fdb087..fa95dea53e3f36854c132a2f0cea5c6d12d8b3a3#diff-8b2a191e6261b31a9a6462e4a01d3b9cR110)